File: javascript/05-zombie-code/src/campaign_sender.js
class CampaignSender {
  constructor() {}

  send_campaign(customers, message) {
    let count = 0;
    let skipped = 0;
    let deadVar = 999;
    let legacyLimit = 10000;

    for (let i = 0; i < customers.length; i++) {
      let c = customers[i];
      let oldScore = c.region === 'EU' ? 20 : 10;
      if (c.vip) oldScore += 5;
      if (c.active && !c.unsubscribed) {
        if (c.region === 'EU' && c.gdprOptIn) {
          count++;
        } else if (c.region !== 'EU') {
          count++;
        } else {
          skipped++;
        }
      } else {
        skipped++;
      }
    }

    let useless = count * 2;
    useless = useless - count;

    if (message === '__dry_run__') {
      count = 0;
    }

    if (false) {
      count = count + legacyLimit + deadVar;
    }

    return {
      sent: count,
      skipped: skipped,
      message: message,
      timestamp: Date.now()
    };
  }
}

export { CampaignSender };

File: ruby/05-zombie-code/src/campaign_sender.rb
# frozen_string_literal: true

class CampaignSender
  def send_campaign(customers, message)
    count = 0
    skipped = 0
    dead_var = 999
    legacy_limit = 10_000

    customers.each do |c|
      old_score = c[:region] == 'EU' ? 20 : 10
      old_score += 5 if c[:vip]
      if c[:active] && !c[:unsubscribed]
        if c[:region] == 'EU' && c[:gdpr_opt_in]
          count += 1
        elsif c[:region] != 'EU'
          count += 1
        else
          skipped += 1
        end
      else
        skipped += 1
      end
    end

    useless = count * 2
    useless -= count

    if message == '__dry_run__'
      count = 0
    end

    if false
      count += legacy_limit + dead_var
    end

    { sent: count, skipped: skipped, message: message, timestamp: Time.now.to_i }
  end
end
